lib/oli_web/plugs/set_user.ex

```diff
@@ -21,13 +22,23 @@ defmodule Oli.Plugs.SetCurrentUser do
     if author = Pow.Plug.current_user(conn, pow_config) do
       cond do
         current_author = Repo.get(Author, author.id) ->
+          is_community_admin =
+            current_author.id
+            |> Accounts.list_admin_communities()
```

Comment:
Given that this plug executes on virtually every end user request, we need to be careful here about keeping this lean.  This change introduces a second database read after the retrieval of the author in order to fetch all admin communities that the user belongs to.  It would be optimal, instead, to replace the `Repo.get(Author, author.id)` with a call to a new function that retrieves both the Author and pulls in a count of the number of communities that they are an admin to (say, perhaps, as a virtual field in the Author schema).  In this manner, we avoid the latency of multiple, sequential database reads.

Reply:
I like it! Thanks